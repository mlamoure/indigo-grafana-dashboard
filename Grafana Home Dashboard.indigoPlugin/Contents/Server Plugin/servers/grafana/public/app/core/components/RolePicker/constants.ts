export const MENU_MAX_HEIGHT = 300; // max height for the picker's dropdown menu
