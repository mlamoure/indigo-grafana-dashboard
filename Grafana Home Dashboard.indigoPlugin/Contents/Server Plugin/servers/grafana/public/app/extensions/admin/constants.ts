export const LICENSED_ROLE_FILTER = 'licensedRole';
