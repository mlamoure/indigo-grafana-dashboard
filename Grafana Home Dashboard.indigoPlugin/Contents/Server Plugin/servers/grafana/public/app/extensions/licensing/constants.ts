// Thresholds for license limits, in percents
export const WARNING_RATE = 80;
export const ERROR_RATE = 100;

// these mirror the definition in types.go
export const NOTLOADED = 0;
export const VALID = 1;
export const LOADED = 2;
export const INVALID = 3;
export const NOTFOUND = 4;
export const EXPIRED = 5;
export const INVALID_SUBJECT = 6;

export const DISMISS_WARNING_FOR_DAYS = 5;
/* The auto dismiss cannot be too low as it will also trigger the snooze */
export const WARNING_CLOSE_TIMEOUT_SEC = 3600;
export const LICENSE_WARNING_DISMISS_UNTIL_KEY = 'grafana.licence.warning.dismissUntil';

export const LIMIT_BY_ROLE_USERS = 'role_users';
export const LIMIT_BY_USERS = 'users';

export const AWS_MARKEPLACE_ISSUER = 'AWS/Marketplace';
