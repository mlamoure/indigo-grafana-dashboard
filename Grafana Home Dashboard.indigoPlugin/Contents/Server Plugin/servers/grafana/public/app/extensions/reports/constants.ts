import { FormRequiredFields, StepKey } from '../types';

export const defaultReportLogo = '/public/img/grafana_icon.svg';
export const defaultEmailLogo = 'https://grafana.com/static/assets/img/grafana_logo_lockup_ltbg.png';

export const BASE_URL = '/reports';

export const requiredFields: FormRequiredFields = [
  { step: StepKey.SelectDashboard, fields: ['dashboardName'] },
  { step: StepKey.Share, fields: ['name', 'recipients'] },
];
