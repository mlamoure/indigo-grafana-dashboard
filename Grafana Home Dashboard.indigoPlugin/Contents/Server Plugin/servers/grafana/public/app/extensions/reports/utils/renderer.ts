// Get renderer version from its config field, return 1 by default
import { config } from '@grafana/runtime';

export const getRendererMajorVersion = (): number | null => {
  const version = config.rendererVersion;
  if (version === '') {
    return null;
  }

  const semverRegex =
    /^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$/;
  const res = semverRegex.exec(version);

  let majorVersionStr = '';
  if (res && res.length > 1) {
    majorVersionStr = res[1];
  }

  if (majorVersionStr === '') {
    return 1;
  }

  let majorVersion = parseInt(majorVersionStr, 10);
  if (isNaN(majorVersion)) {
    majorVersion = 1;
  }

  return majorVersion;
};
