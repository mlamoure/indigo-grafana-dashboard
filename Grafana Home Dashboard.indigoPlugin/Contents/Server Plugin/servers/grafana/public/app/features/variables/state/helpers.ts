import { combineReducers } from '@reduxjs/toolkit';

import { NEW_VARIABLE_ID } from './types';
import { VariableHide, VariableModel } from '../../templating/types';
import { variablesReducer, VariablesState } from './variablesReducer';
import { optionsPickerReducer } from '../pickers/OptionsPicker/reducer';
import { variableEditorReducer } from '../editor/reducer';
import { locationReducer } from '../../../core/reducers/location';
import { VariableAdapter } from '../adapters';
import { dashboardReducer } from 'app/features/dashboard/state/reducers';

export const getVariableState = (
  noOfVariables: number,
  inEditorIndex = -1,
  includeEmpty = false
): Record<string, VariableModel> => {
  const variables: Record<string, VariableModel> = {};

  for (let index = 0; index < noOfVariables; index++) {
    variables[index] = {
      id: index.toString(),
      type: 'query',
      name: `Name-${index}`,
      hide: VariableHide.dontHide,
      index,
      label: `Label-${index}`,
      skipUrlSync: false,
    };
  }

  if (includeEmpty) {
    variables[NEW_VARIABLE_ID] = {
      id: NEW_VARIABLE_ID,
      type: 'query',
      name: `Name-${NEW_VARIABLE_ID}`,
      hide: VariableHide.dontHide,
      index: noOfVariables,
      label: `Label-${NEW_VARIABLE_ID}`,
      skipUrlSync: false,
    };
  }

  return variables;
};

export const getVariableTestContext = <Model extends VariableModel>(
  adapter: VariableAdapter<Model>,
  variableOverrides: Partial<Model> = {}
) => {
  const defaultVariable = {
    ...adapter.initialState,
    id: '0',
    index: 0,
    name: '0',
  };

  const initialState: VariablesState = {
    '0': { ...defaultVariable, ...variableOverrides },
  };

  return { initialState };
};

export const getRootReducer = () =>
  combineReducers({
    location: locationReducer,
    dashboard: dashboardReducer,
    templating: combineReducers({
      optionsPicker: optionsPickerReducer,
      editor: variableEditorReducer,
      variables: variablesReducer,
    }),
  });

export const getTemplatingRootReducer = () =>
  combineReducers({
    templating: combineReducers({
      optionsPicker: optionsPickerReducer,
      editor: variableEditorReducer,
      variables: variablesReducer,
    }),
  });

export const getTemplatingAndLocationRootReducer = () =>
  combineReducers({
    templating: combineReducers({
      optionsPicker: optionsPickerReducer,
      editor: variableEditorReducer,
      variables: variablesReducer,
    }),
    location: locationReducer,
  });
