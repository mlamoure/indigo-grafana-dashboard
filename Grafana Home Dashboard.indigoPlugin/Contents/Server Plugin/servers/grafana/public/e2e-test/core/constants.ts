export const constants = {
  baseUrl: process.env.BASE_URL || 'http://localhost:3000',
  chromiumRevision: '650629',
  screenShotsTruthDir: './public/e2e-test/screenShots/theTruth',
  screenShotsOutputDir: './public/e2e-test/screenShots/theOutput',
};
