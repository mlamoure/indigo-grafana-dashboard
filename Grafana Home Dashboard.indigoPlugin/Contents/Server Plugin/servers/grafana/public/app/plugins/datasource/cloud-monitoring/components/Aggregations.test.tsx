import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { Segment } from '@grafana/ui';
import { Aggregations, Props } from './Aggregations';
import { ValueTypes, MetricKind } from '../constants';
import { TemplateSrvStub } from 'test/specs/helpers';

const props: Props = {
  onChange: () => {},
  // @ts-ignore
  templateSrv: new TemplateSrvStub(),
  metricDescriptor: {
    valueType: '',
    metricKind: '',
  },
  crossSeriesReducer: '',
  groupBys: [],
  children: renderProps => <div />,
  templateVariableOptions: [],
};

describe('Aggregations', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Aggregations {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('options', () => {
    describe('when DOUBLE and GAUGE is passed as props', () => {
      const nextProps = {
        ...props,
        metricDescriptor: {
          valueType: ValueTypes.DOUBLE,
          metricKind: MetricKind.GAUGE,
        },
      };

      it('should not have the reduce values', () => {
        const wrapper = shallow(<Aggregations {...nextProps} />);
        const { options } = wrapper.find(Segment).props() as any;
        const [, aggGroup] = options;

        expect(aggGroup.options.length).toEqual(11);
        expect(aggGroup.options.map((o: any) => o.value)).toEqual(
          expect.not.arrayContaining(['REDUCE_COUNT_TRUE', 'REDUCE_COUNT_FALSE'])
        );
      });
    });

    describe('when MONEY and CUMULATIVE is passed as props', () => {
      const nextProps = {
        ...props,
        metricDescriptor: {
          valueType: ValueTypes.MONEY,
          metricKind: MetricKind.CUMULATIVE,
        },
      };

      it('should have the reduce values', () => {
        const wrapper = shallow(<Aggregations {...nextProps} />);
        const { options } = wrapper.find(Segment).props() as any;
        const [, aggGroup] = options;

        expect(aggGroup.options.length).toEqual(10);
        expect(aggGroup.options.map((o: any) => o.value)).toEqual(expect.arrayContaining(['REDUCE_NONE']));
      });
    });
  });
});
