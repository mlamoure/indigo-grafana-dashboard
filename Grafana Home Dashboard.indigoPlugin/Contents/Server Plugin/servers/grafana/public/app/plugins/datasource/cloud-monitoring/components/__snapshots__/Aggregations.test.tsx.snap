// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Aggregations renders correctly 1`] = `
Array [
  <div
    className="gf-form-inline"
  >
    <label
      className="gf-form-label query-keyword width-9"
    >
      Aggregation
    </label>
    <div
      className="gf-form"
      onClick={[Function]}
    >
      <a
        className="gf-form-label query-part query-placeholder"
      >
        Select Reducer
      </a>
    </div>
    <div
      className="gf-form gf-form--grow"
    >
      <label
        className="gf-form-label gf-form-label--grow"
      >
        <a
          onClick={[Function]}
        >
          <div
            className="css-1cvxpvr"
          >
            <svg
              className="css-sr6nr"
              fill="currentColor"
              height={16}
              viewBox="0 0 24 24"
              width={16}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14.83,11.29,10.59,7.05a1,1,0,0,0-1.42,0,1,1,0,0,0,0,1.41L12.71,12,9.17,15.54a1,1,0,0,0,0,1.41,1,1,0,0,0,.71.29,1,1,0,0,0,.71-.29l4.24-4.24A1,1,0,0,0,14.83,11.29Z"
              />
            </svg>
          </div>
           Advanced Options
        </a>
      </label>
    </div>
  </div>,
  <div />,
]
`;
